src/Sylius/Bundle/CartBundle/EventListener/ClearAdjustmentsOnEmptyCartListener.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\CartBundle\EventListener;
+
+use Sylius\Component\Cart\Model\CartInterface;
+use Symfony\Component\EventDispatcher\GenericEvent;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class ClearAdjustmentsOnEmptyCartListener
+{
+    /**
+     * @param GenericEvent $event
+     */
+    public function clearAdjustmentsOnEmptyCart(GenericEvent $event)
```

Comment:
Or, if you prefer to name it after it purpose, try to change this method name, to not duplicate name of listener ;)